pkg/vz/vm_darwin.go

```diff
@@ -0,0 +1,485 @@
+		return nil, err
+	}
+
+	if err = attachDisplay(driver, vmConfig); err != nil {
+		return nil, err
+	}
+
+	if err = attachConsole(driver, vmConfig); err != nil {
+		return nil, err
+	}
+
+	if err = attachFolderMounts(driver, vmConfig); err != nil {
+		return nil, err
+	}
+
+	if err = attachOtherDevices(driver, vmConfig); err != nil {
+		return nil, err
+	}
+
+	validated, err := vmConfig.Validate()
+	if !validated || err != nil {
+		return nil, err
+	}
+
+	return vz.NewVirtualMachine(vmConfig)
+}
+
+func createInitialConfig(driver *driver.BaseDriver) (*vz.VirtualMachineConfiguration, error) {
+	efiVariableStore, err := getEFI(driver)
+	if err != nil {
+		return nil, err
+	}
+
+	bootLoader, err := vz.NewEFIBootLoader(vz.WithEFIVariableStore(efiVariableStore))
+	if err != nil {
+		return nil, err
+	}
+
+	bytes, err := units.RAMInBytes(*driver.Yaml.Memory)
+	if err != nil {
+		return nil, err
+	}
+
+	vmConfig, err := vz.NewVirtualMachineConfiguration(
+		bootLoader,
+		uint(*driver.Yaml.CPUs),
+		uint64(bytes),
+	)
+	if err != nil {
+		return nil, err
+	}
+	return vmConfig, nil
+}
+
+func attachPlatformConfig(driver *driver.BaseDriver, vmConfig *vz.VirtualMachineConfiguration) error {
+	machineIdentifier, err := getMachineIdentifier(driver)
+	if err != nil {
+		return err
+	}
+
+	platformConfig, err := vz.NewGenericPlatformConfiguration(vz.WithGenericMachineIdentifier(machineIdentifier))
+	if err != nil {
+		return err
+	}
+	vmConfig.SetPlatformVirtualMachineConfiguration(platformConfig)
+	return nil
+}
+
+func attachSerialPort(driver *driver.BaseDriver, config *vz.VirtualMachineConfiguration) error {
+	path := filepath.Join(driver.Instance.Dir, filenames.SerialLog)
+	serialPortAttachment, err := vz.NewFileSerialPortAttachment(path, false)
+	if err != nil {
+		return err
+	}
+	consoleConfig, err := vz.NewVirtioConsoleDeviceSerialPortConfiguration(serialPortAttachment)
+	config.SetSerialPortsVirtualMachineConfiguration([]*vz.VirtioConsoleDeviceSerialPortConfiguration{
+		consoleConfig,
+	})
+	return err
+}
+
+func attachNetwork(driver *driver.BaseDriver, vmConfig *vz.VirtualMachineConfiguration, networkConn *os.File) error {
+	//slirp network using gvisor netstack
+	fileAttachment, err := vz.NewFileHandleNetworkDeviceAttachment(networkConn)
+	if err != nil {
+		return err
+	}
+	err = fileAttachment.SetMaximumTransmissionUnit(1500)
+	if err != nil {
+		return err
+	}
+	networkConfig, err := vz.NewVirtioNetworkDeviceConfiguration(fileAttachment)
+	if err != nil {
+		return err
+	}
+	mac, err := net.ParseMAC(limayaml.MACAddress(driver.Instance.Dir))
+	if err != nil {
+		return err
+	}
+	address, err := vz.NewMACAddress(mac)
+	if err != nil {
+		return err
+	}
+	networkConfig.SetMACAddress(address)
+
+	configurations := []*vz.VirtioNetworkDeviceConfiguration{
+		networkConfig,
+	}
+	//Other networks
+	for _, network := range driver.Yaml.Networks {
+		//TODO - Handle NAT network type
+		if network.Socket != "" {
```

Comment:
@AkihiroSuda 
For NAT network type, shall i go ahead with `network.NAT: true/false` or `network.socket: NAT` ?

Reply:
What do you mean by "NAT"?

Reply:
Anyway probably it should be a separate PR

Reply:
For host to guest network, the below code uses https://developer.apple.com/documentation/virtualization/vznatnetworkdeviceattachment provided from virtualization.framework itself.

This doesn't require sudo access.

Reply:
Would it be possible to  just use vznat by default and drop gvisor netstack ?

If vznat has to be optional, probably it should be like `network.VZNAT: true/false`

Reply:
Yes that should work but with vzNAT there will be other issues like VPN won't be working. 

Also i faced issues with calls to host dns resolver via UDP didn't work

Reply:
Thanks, let's keep using gvisor netstack by default, and revisit VZNAT in a separate PR